packages/integration-tests/src/collections/CustomReference.test.ts

```diff
@@ -0,0 +1,80 @@
+import { EntityManager } from "joist-orm";
+import { knex, numberOfQueries, resetQueryCount } from "../setupDbTests";
+import { Author, Book, BookReview } from "../entities";
+import { insertAuthor, insertBook, insertBookReview, insertPublisher } from "../entities/factories";
+
+describe("CustomReference", () => {
+  it("can load a reference", async () => {
+    await insertAuthor({ first_name: "f" });
+    await insertBook({ title: "t", author_id: 1 });
+    await insertBookReview({ rating: 5, book_id: 1 });
+
+    const em = new EntityManager(knex);
+    const review = await em.load(BookReview, "1");
+    const author = await review.author.load();
+    expect(author.firstName).toEqual("f");
+  });
+
+  it("can populate a reference", async () => {
+    await insertAuthor({ first_name: "f" });
+    await insertBook({ title: "t", author_id: 1 });
+    await insertBookReview({ rating: 5, book_id: 1 });
+
+    const em = new EntityManager(knex);
+    const review = await em.load(BookReview, "1", "author");
+    expect(review.author.get.firstName).toEqual("f");
+  });
+
+  it("can set a reference", async () => {
+    const em = new EntityManager(knex);
+    const author = em.create(Author, { firstName: "a1" });
+    const book = em.createUnsafe(Book, { title: "t1" });
+    const review = em.createUnsafe(BookReview, { book, rating: 5 });
+    review.author.set(author);
+    await em.flush();
+
+    const rows = await knex.select("*").from("books");
+    expect(rows[0].author_id).toEqual(1);
+  });
+
+  it("can set a reference through setFromOpts", async () => {
+    const em = new EntityManager(knex);
+    const author = em.create(Author, { firstName: "a1" });
+    const book = em.createUnsafe(Book, { title: "t1" });
+    const review = em.createUnsafe(BookReview, { book, rating: 5, author } as any);
```

Comment:
Huh. That's fancy. It's not technically in the Opts type, but the `as any` makes it work?

Reply:
yup